src/HDF5.jl

```diff
@@ -1381,8 +1380,8 @@ end
 # Create datasets and attributes with "native" types, but don't write the data.
 # The return syntax is: dset, dtype = d_create(parent, name, data; properties...)
 for (privatesym, fsym, ptype) in
-    ((:_d_create, :d_create, Union{HDF5File, HDF5Group}),
-     (:_a_create, :a_create, Union{HDF5File, HDF5Group, HDF5Dataset, HDF5Datatype}))
+    ((:_d_create, :d_create, Union{File,Group}),
```

Comment:
We might want to unroll these to improve precompile speeds?

Reply:
I agree, but I think it'd be OK to wait on for another PR.

Reply:
Definitely new PR but thought I'd mention it.